compiler/rustc_trait_selection/src/opaque_types.rs

```diff
@@ -1051,8 +1051,11 @@ impl<'a, 'tcx> Instantiator<'a, 'tcx> {
             item_bounds.iter().map(|(bound, _)| bound.subst(tcx, substs)).collect();
 
         let param_env = tcx.param_env(def_id);
-        let InferOk { value: bounds, obligations } =
-            infcx.partially_normalize_associated_types_in(span, self.body_id, param_env, bounds);
+        let InferOk { value: bounds, obligations } = infcx.partially_normalize_associated_types_in(
+            ObligationCause::misc(span, self.body_id),
```

Comment:
I wonder if we could (maybe later?) introduce a better `ObligationCauseCode` here... 🤔

Reply:
I believe there's a lot of ongoing refactoring of the TAIT code right now, so I think it would be best to leave this for later.

Reply:
Works for me. I want to audit `MiscObligation` again at some point.